fab/sampling_methods/base.py

```diff
@@ -69,7 +69,7 @@ def create_point(x: torch.Tensor, log_q_fn: LogProbFunc, log_p_fn: LogProbFunc,
     else:
         # Use log_q_x if we already have it, otherwise calculate it.
         log_q_x = log_q_x if log_q_x is not None else log_q_fn(x)
-        return Point(x=x, log_q=log_q_x, log_p=log_p_fn(x))
+        return Point(x=x, log_q=log_q_x.detach(), log_p=log_p_fn(x).detach())
```

Comment:
No longer re-using log_q_x in the loss function (as if it contains NaNs then this causes numerical instability)